### DISCUSSION TITLE
Is the Emotion project still active?

### DISCUSSION THREAD

**Discussion by philp** [emotion-js/emotion#2443] [General]
Hi there! I arrived at this project after learning about it through Material UI. Today, I thought I'd give it a spin with a new side project, but unfortunately ran into a compilation exception (detailed issue submitted like a responsible dev!)

Looking at the repo, and the number of issues awaiting triage, it doesn't look like there's been much activity recently. It looks like a great library, but I'm curious to know what the current state of play is, before I invest any more time in trying to et it working?

Thanks in advance!

**Comment by ivancuric** [comment]
I think the silence speaks volumes.

**Reply by Andarist** [reply]
I've changed my job at the beginning of July and didn't have that much time to handle the issues as I've used to. That doesn't mean that the project is dead and I'm very much monitoring issues and taking care of whatever is critical.

The vast majority of the created issues are, quite frankly, support tickets. Functionally - there are no big issues that await resolution. Handling a single ticket like this often takes more than just a couple of seconds and it's hard to justify jumping onto everything that pops up into my notifications. Even if I would really like to do so, and believe me - I would like to. I like to think that when I had time I've provided way more help per issue than the average.

This all is mostly free work though, I have a day job and 2 small children, I'm just a single guy and time is not an unlimited resource (unfortunately). I really have to put some priority on such things, especially that this is not the only project that I'm taking care.

The project is also open-source for a reason - I strongly believe that `node_modules` is as much of "my code" as whatever I write for my day job, in private repositories etc. This is, in fact, how I end up maintaining so many projects - if I use a project daily I spot bugs, want some new features etc. So I just go and dive into node_modules to see how I can improve things for myself (and others). This is an option.

I really don't want to be rude - but I'm getting really bitter when I read stuff like this about what I provide **for free to everyone**. Even if the project would be "dead" - you can always take it, volunteer to become a maintainer, or just fork it. That's the beauty and the curse of OSS in its current form.

**Reply by philp** [reply]
I really didn't intend for my query to come across as an "is this project dead?" snipe. I really was just very curious to understand whether it was actively maintained, particularly as I'd spotted that Material UI appears to have adopted it as the default styling library.

I'm always keen to contribute PRs when I find something I can fix. But on this occasion, I was struggling to even get the project to build, so poking around wasn't much of an option. That's one of the reasons I reached out. I hoped my enquiry was polite; sorry if it didn't come across that way.

**Reply by Andarist** [reply]
No worries - sorry if my response sounded a little bit harsh. I know that, for the most part, individuals don't have bad intentions. I didn't care about that in the past that much but since I really have way less free time nowadays it started to bother me more. I don't want to bash on anyone in particular - it's more like venting for me, screaming into the universe, raising awareness of the problem in a hope that maybe people notice it and something gets changed... someday.

In other words - ask your employers to start sponsoring OSS. And it's really not about me - they don't have to sponsor my work, I would be happy to know that others in the community get compensated somehow.

**Reply by Andarist** [reply]
Oh, as to MUI - I'm in touch with the MUI team, been collaborating on some PRs to Emotion with them, etc. It's just that there is nothing critical to be taken care of here at the moment and I don't have time (unfortunately) to answer all support questions here. So I just prioritize - be aware that I'm keeping an eye on everything that happens in this repository and if something is urgent then I take care of it. So even if issues are "piling up"... they really are not. I just don't want to rudely close them without an answer or redirect them to SO.

**Comment by ivancuric** [comment]
Sorry if I came off as being dismissive. I can't even imagine what it must
be to support an open source project.

I was mainly worried if there are any other maintainers active, since a
single person can't usually handle it all. The discussions in GitHub, Slack
and Spectrum seem pretty dead as well.

**Reply by Andarist** [reply]
I agree it's not a one-man's job - but it takes a volunteer to help out with that. And since I know how it is - I definitely won't seek out any people on purpose unless somebody just volunteers themselves.
